frontend/src/widgets/NavBar/ui/NavLink.tsx

```diff
@@ -0,0 +1,39 @@
+import { Link, useResolvedPath, useMatch } from 'react-router-dom';
+import styled from 'styled-components';
+
+interface NavLinkProps {
+	to: string;
+	title: string;
+}
+
+export function NavLink({ to, title }: NavLinkProps) {
+	const resolved = useResolvedPath(to);
+	const match = useMatch({ path: resolved.pathname, end: true });
+
+	return (
+		<Root isActive={Boolean(match)}>
+			<StyledLink to={to}>{title}</StyledLink>
+		</Root>
+	);
+}
+
+const Root = styled.li<{ isActive: boolean }>`
+	width: 100%;
+	transition: all 0.3s;
+	border: 1px solid ${({ isActive }) => (isActive ? 'gray' : 'black')};
+	background-color: ${({ isActive }) => (isActive ? '#e7e2e2' : 'white')};
```

Comment:
Do you want to use props or data-attributes?
If you want to use props, prefer to use transient props `$isActive` by styled-components documentation